rpc_status.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# RPC username
+USER=""
+
+# RPC password
+PASSWORD=""
+
+# RPC domain with authentication method
+# Example: "perun.cesnet.cz/krb"
+DOMAIN=""
+
+# Valid userId - This id will be used in getUSerById call
+USER_ID=""
+
+# eduPersonPrincipalName which will be searched
+URL="https://${DOMAIN}/rpc/json/usersManager/getUserById?id=${USER_ID}"
+
+START_TIME=$(date +%s%N)
+RPC_RESULT=$(timeout 10 curl --user ${USER}:${PASSWORD} ${URL} 2>&1)
+END_TIME=$(date +%s%N)
+TOTAL_TIME=$(echo "scale=4;$(expr ${END_TIME} - ${START_TIME}) / 1000000000" | bc -l)
+if [[ $RPC_RESULT == *\"id\":${USER_ID}*  ]]; then
+    echo "0 rpc_status total_time=${TOTAL_TIME} OK"
+else
+    echo "2 rpc_status total_time=${TOTAL_TIME} ${RPC_RESULT}"
```

Comment:
Just a question of concern: is the return code 1 reserved for anything special as you have used 2 instead?

Reply:
The return code 1 is reserved for Nagios state: WARNING